server/router.go

```diff
@@ -161,30 +159,20 @@ func (router *router) HandleMessage(message *protocol.Message) error {
 		return &PermissionDeniedError{message.UserID, auth.WRITE, message.Path}
 	}
 
-	lenMessage := int64(len(message.Bytes()))
-	mTotalMessagesIncomingBytes.Add(lenMessage)
+	// for a new locally-generated message, we need to generate a new message-ID
+	var nodeID int
+	if router.cluster != nil {
+		nodeID = router.cluster.Config.ID
+	}
 
-	msgPathPartition := message.Path.Partition()
-	if router.cluster == nil || (router.cluster != nil && message.NodeID == router.cluster.Config.ID) {
-		// for a new locally-generated message, we need to generate a new message-ID
-		txCallback := func(msgId uint64) []byte {
-			message.ID = msgId
-			message.Time = time.Now().Unix()
-			return message.Bytes()
-		}
-		if err := router.messageStore.StoreTx(msgPathPartition, txCallback); err != nil {
-			logger.WithError(err).WithField("msgPartition", msgPathPartition).Error("Error storing new local message in partition")
-			mTotalMessageStoreErrors.Add(1)
-			return err
-		}
+	mTotalMessagesIncomingBytes.Add(int64(len(message.Bytes())))
+	if size, err := router.messageStore.StoreMessage(message, nodeID); err != nil {
```

Comment:
@Bogh 
StoreMessage returns now a message size ? 
I think we know the message size already, before calling this func.
If StoreMessage is not called elsewhere, I suggest to change its signature to return only a possible "error".